docs/source/examples/beam.py

```diff
@@ -47,6 +47,8 @@ def __init__(self, N=4, **kwargs):
         theta_eq[0] = (th[0] >= th_base)  # base boundary condition
         displ_eq = (w >= w.left + 0.5*dx*(th + th.left))
         displ_eq[0] = (w[0] >= w_base)
+        for constraint in [shear_eq, moment_eq, theta_eq, displ_eq]:
+            constraint.reset_varkeys()
```

Comment:
fair -- could `__setitem__` do something more subtle, like adding the new variables introduced by whatever is passed in, but not calling `reset_varkeys`?

Another option is to try to rewrite Beam in a better way that doesn't mutate array constraints